src/service.js

```diff
@@ -188,8 +207,16 @@ class Service {
     // We need this shitty hack because update doesn't return
     // a promise properly when runValidators is true. WTF!
     try {
+      let query = Object.assign({}, params.query);
+
+      // If params.query.$populate was provided, remove it
+      // from the query sent to mongoose.
+      if (query.$populate) {
+        delete query.$populate;
```

Comment:
Needs to be `query = omit(query, '$populate')` as to not modify the original object.